### PR TITLE
created new test DeleteProductFromCartTest

### PR DESCRIPTION
**Pull request by KhraschukZahar**

### CODE REVIEW

File: src/test/java/base/AbstractBaseTest.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public void openBrowser() {
 
     @AfterMethod
     public void closeBrowser() {
-        driver.quit();
+//        driver.quit();
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/base/AbstractBaseTest.java:32]: resolved

File: src/test/java/pages/CartPage.java [hunk 2]
```diff
@@ -42,4 +48,9 @@ public List<String> getInfoAboutProductOnCart(String cart_size,String price) {
 
         return infoAboutShoesOnCart;
     }
+    public boolean IsThereAreNoItemsInYourCartVisibleTitle(){
+        return driver.findElement(By.xpath("//span[text()='THERE ARE NO ITEMS IN YOUR CART']"))
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/CartPage.java:52]: resolved

File: src/test/java/tests/AddToCartTest.java [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 public class AddToCartTest extends AbstractBaseTest {
 
-    private static final String SIZE = "2.5";
+    protected static final String SIZE = "2.5";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/tests/AddToCartTest.java:12]: resolved

File: src/test/java/tests/DeleteProductFromCartTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import pages.BrowseProductsPage;
+import pages.CartPage;
+import pages.HomePage;
+import pages.ProductPage;
+
+import static tests.AddToCartTest.SIZE;
+
+public class DeleteProductFromCartTest extends AbstractBaseTest {
+
+    @Test
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/tests/DeleteProductFromCartTest.java:15]: resolved

File: src/test/java/tests/DeleteProductFromCartTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        browseProductsPage.selectPage();
+
+        productPage.selectSizeDropdown();
+        productPage.selectSize(SIZE);
+        productPage.addToCart();
+        productPage.proceedToCheckout();
+
+        cartPage.removeItem();
+
+        Assert.assertTrue(driver.getCurrentUrl().contains("OrderItemDisplay"));
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/tests/DeleteProductFromCartTest.java:34]: resolved

File: src/test/java/tests/DeleteProductFromCartTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        Assert.assertTrue(cartPage.IsThereAreNoItemsInYourCartVisibleTitle());
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/tests/DeleteProductFromCartTest.java:35]: resolved

File: src/test/java/tests/CartTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        productPage = new ProductPage(driver);
+        browseProductsPage = new BrowseProductsPage(driver);
+        cartPage = new CartPage(driver);
+
+        homePage.goToCatalogueShopNow();
+        browseProductsPage.selectPage();
+        productPage.selectSizeDropdown();
+        productPage.selectSize(SIZE);
+        productPage.addToCart();
+    }
```
Comment (Goldbe54) [inline src/test/java/tests/CartTest.java:43]: We do not use the before method in tests. Talk to your project mentor about it

**Review by Goldbe54 — APPROVED** [review]

File: src/test/java/tests/CartTest.java (hunk 7 above)
Comment (KhraschukZahar) [reply src/test/java/tests/CartTest.java:43]: resolved

File: src/test/java/pages/CartPage.java [hunk 8]
```diff
@@ -11,11 +12,15 @@
...
-    protected final String SIZE_SHOES_FIELD_CART = "//dd[contains(text(),'%s')]";
+    private final String NAME_SHOES_FIELD_CART = "//h3[@class='item-detail-info-name item-name-js']";
+    private final String PRICE_SHOES_FIELD_CART = "//dd[contains(text(),'%s')]";
+    private final String SIZE_SHOES_FIELD_CART = "//dd[contains(text(),'%s')]";
+    private final String REMOVE_ITEM = "//div[@class='checkout-item-detail-edit item-detail-edit']//a[@data-action='delete']";
+    private final String CHECK_CART_ITEM = "//span[text()='THERE ARE NO ITEMS IN YOUR CART']";
+    private final String AFTER_DELETE_MESSAGE = "THERE ARE NO ITEMS IN YOUR CART";
     protected final String CHECKOUT_BUTTON = "//a[@class='button primary btn-checkout-js ']";
 
+
```
Comment (Goldbe54) [inline src/test/java/pages/CartPage.java:23]: delete empty line

File: src/test/java/pages/CartPage.java [hunk 9]
```diff
@@ -43,6 +52,10 @@ public List<String> getInfoAboutProductOnCart(String cart_size,String price) {
 
         return infoAboutShoesOnCart;
     }
+    public boolean IsThereAreNoItemsInYourCartVisibleTitle(){
```
Comment (Goldbe54) [inline src/test/java/pages/CartPage.java:55]: add empty line

**Review by Goldbe54 — APPROVED** [review]

File: src/test/java/pages/CartPage.java (hunk 9 above)
Comment (KhraschukZahar) [reply src/test/java/pages/CartPage.java:55]: resolved

**Review by SoniaMiach — APPROVED** [review]

File: src/test/java/pages/CartPage.java (hunk 8 above)
Comment (KhraschukZahar) [reply src/test/java/pages/CartPage.java:23]: resolved

**Pull request merged**